### ISSUE TITLE
wrapdims(::DataFrame) produces incorrect results when not all key combinations are present

### ISSUE THREAD

**Issue by fredcallaway** [mcabbott/AxisKeys.jl#105]
```julia
df = DataFrame([
    (x="a", y="a", z=1),
    (x="a", y="b", z=2),
    (x="b", y="b", z=3),
])
wrapdims(df, :z, :x, :y)
```

The result varies, but here's one example
```
                 ("a")  ("b")
  ("a")           1      2
  ("b")  4615108224      3
```

I think the matrix should be initialized with an undefined value rather than an arbitrary value (maybe `missing`)?

**Comment by fredcallaway** [comment]
Sorry, I should have read the documentation better. I think the `default` argument should be missing by default. The current default behavior can lead to errors from careless users like myself (and let's be honest, there are many of us).

**Comment by rofinn** [comment]
The issue with making the default `missing` is that this introduces unnecessary type complexity in many cases. If there are no `missing` values the returned element type is still `Union{T, Missing}` which may break downstream code. I'm not particularly bothered either way,  that's the main reason we didn't make it the default.

**Comment by fredcallaway** [comment]
Yeah, I totally see why you would want to not use missing if you know there are no missing values. But I guess I think it's better to fail obviously (with an error) than to produce incorrect values. One could even make an argument for requiring the default value be passed explicitly.

The minimal action would be to add a more salient warning to the documentation (both the README and the docstring). I can draft that if you like.

**Comment by rofinn** [comment]
> default value be passed explicitly

Okay, yeah, that's probably the best path forward.
